File: Bridge/cpp-source/wheels/Gamepad.cpp
#include <utility>

//
// Created by Pavel Akhtyamov on 2019-03-26.
//

#include <Gamepad.h>


void Gamepad::PushWheel() {
  axle_->Beep();
}

void Gamepad::PullWheel() {
}

void Gamepad::PressGasPedal() {
  axle_->Accelerate(0.1);
}

void Gamepad::PressBreakPedal() {
  axle_->Brake(0.1);
}

void Gamepad::PressLeftButton() {
  axle_->MoveLeft(0.1);
}

void Gamepad::PressRightButton() {
  axle_->MoveRight(0.1);
}

void Gamepad::MoveRightWheel(float angle) {}

void Gamepad::MoveLeftWheel(float angle) {}

void Gamepad::PressUpButton() {
  axle_->MoveDown(0.1);
}

void Gamepad::PressDownButton() {
  axle_->MoveUp(0.1);
}

Gamepad::Gamepad(std::unique_ptr<Axle> ax) : SteeringWheel(std::move(ax)) {

}

void Gamepad::PrintInfo() {
  axle_->PrintInfo();
}

File: Bridge/python-source/wheels/Gamepad.py
from __future__ import annotations

from .SteeringWheel import SteeringWheel


class Gamepad(SteeringWheel):
    """Геймпад: кнопки вместо руля и педалей.

    Обрати внимание на кнопки вверх и вниз: они намеренно переставлены местами
    относительно осей. Это настройка органа управления, и техника про неё
    ничего не знает.
    """

    STEP = 0.1

    def push_wheel(self) -> None:
        self._axle.beep()

    def pull_wheel(self) -> None:
        """У геймпада нет руля, тянуть нечего."""

    def press_gas_pedal(self) -> None:
        self._axle.accelerate(self.STEP)

    def press_brake_pedal(self) -> None:
        self._axle.brake(self.STEP)

    def press_left_button(self) -> None:
        self._axle.move_left(self.STEP)

    def press_right_button(self) -> None:
        self._axle.move_right(self.STEP)

    def press_up_button(self) -> None:
        self._axle.move_down(self.STEP)

    def press_down_button(self) -> None:
        self._axle.move_up(self.STEP)
